### PR TITLE
Clarification on the occurences of TLV records with the same type

### LINKED ISSUE #776: BOLT 2: Clarifying requirements for multiple TLV record occurrences of the same type
Currenlty nothing in the BOLT-2 specification <https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format> prevents from using multiple TLV records of the same type inside message TLV stream. However, it is unclear whether this is an intentional feature – or just a bug of underspecification. I propose to clarify this point. I am prepared to write a PR for this, but not sure what is the original design intention so will wait for your clarification. Also, if this is the intentional feature, it is than unclear why [Init](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#the-init-message) message takes an array of `chain_hash` as a single TLV record.

### PR DESCRIPTION
**Pull request by dr-orlovsky**
Closes #776 as discussed within the issue

### CODE REVIEW

File: 01-messaging.md [hunk 1]
```diff
@@ -192,6 +194,21 @@ things, enables the following optimizations:
...
+have multiple records. The first defeats the purpose of the proposal, and the
+latter is cumbersome in practice and would further fragment the type space
+beyond the existing even/odd rule.
+
+A minor reasons for strictly-unique types is that permitting multiple of the
+same type is less efficient on the wire due to the overhead of redundant
+type-length pairs. It is also less efficient when allocating memory for the
+decoded objects, e.g. one can't allocate the exact size of a vector upfront and
+will incur more copying overhead when appending each item individually.
+
```
Comment (t-bast) [inline 01-messaging.md:211]: I would remove these paragraphs entirely, they clutter the spec (IMHO).

Comment (dr-orlovsky) [reply 01-messaging.md:211]: Well, I personally find this explanation (when was given it) quire useful; it helps understanding the reasons of particular design. The spec is not that large yet, so having two more paragraphs adding clarity is good. But I am fine to remove them LN community thinks they are unnecessary.

Comment (t-bast) [reply 01-messaging.md:211]: I'll let @cfromknecht chime in, but I think the general feeling when we discussed this during a spec meeting was to remove or condense these paragraphs.

Comment (dr-orlovsky) [reply 01-messaging.md]: Done + rebased

**Comment by dr-orlovsky** [comment]
The paragraph, according to @t-bast critics, is removed. Also, have done a rebase onto master

File: 01-messaging.md [hunk 2]
```diff
@@ -135,7 +135,8 @@ according to the message-specific format determined by `type`.
 ### Requirements
 
 The sending node:
- - MUST order `tlv_record`s in a `tlv_stream` by monotonically-increasing `type`.
+ - MUST order `tlv_record`s in a `tlv_stream` by strictly-increasing `type`,
```
Comment (ysangkok) [inline 01-messaging.md:138]: strictly-increasing is not the same as "unique", which this sentence implies with "i.e."

Comment (t-bast) [reply 01-messaging.md:138]: Strictly increasing does imply unique, doesn't it? (the reverse isn't true)

**Comment by ariard** [comment]
ACK

File: 01-messaging.md (hunk 2 above)
Comment (Roasbeef) [reply 01-messaging.md:138]: Below we have: 
> The strict monotonicity constraint ensures that all `type`s are unique and can at most once. Fields that map to complex objects, e.g. vectors, maps, or	appear at most once

Comment (ysangkok) [reply 01-messaging.md:138]: indeed, but "i.e." implies that they are equivalent in that, as merriam-webster says "it introduces a rewording or a clarification". An implication is not a rewording.

Comment (t-bast) [reply 01-messaging.md:138]: Is your suggestion to change `i.e. MUST not produce` to `hence MUST not produce`?
If so, @dr-orlovsky can you apply this small change and we can merge?

Comment (ysangkok) [reply 01-messaging.md:138]: yes, that is a great idea.

Comment (dr-orlovsky) [reply 01-messaging.md:138]: done!

**Pull request merged**